lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -810,15 +839,25 @@ def __init__(
         on_relation = self._charm.on[self._relation_name]
         self.framework.observe(
             on_relation.relation_created,
-            self._on_relation_change,
+            self._on_consumers_changed,
```

Comment:
I do not recall why we need to observe both `created` and `joined`. Shouldn't `joined` be enough?

Reply:
Largely that Prometheus itself [needs to know in order to enable remote write](https://github.com/canonical/prometheus-k8s-operator/blob/main/src/charm.py#L500). I would also think `joined` would be enough, but... we don't know what we don't know, and I'm happier leaving it exactly as it was for an "emergency" fix which we can evaluate/remove later.